python/ruff/__main__.py

```diff
@@ -35,22 +35,40 @@ def find_ruff_bin() -> str:
 
     # Search for pip-specific build environments.
     #
+    # Expect to find ruff in <prefix>/pip-build-env-<rand>/overlay/bin/ruff
+    # Expect to find a "normal" folder at <prefix>/pip-build-env-<rand>/normal
+    #
     # See: https://github.com/pypa/pip/blob/102d8187a1f5a4cd5de7a549fd8a9af34e89a54f/src/pip/_internal/build_env.py#L87
     paths = os.environ.get("PATH", "").split(os.pathsep)
     if len(paths) >= 2:
-        first, second = os.path.split(paths[0]), os.path.split(paths[1])
-        # Search for both an `overlay` and `normal` folder within a `pip-build-env-{random}` folder. (The final segment
-        # of the path is the `bin` directory.)
+
+        def get_last_three_path_parts(path: str) -> list[str]:
+            """Return a list of up to the last three parts of a path."""
+            parts = []
+
+            while len(parts) < 3:
```

Comment:
I changed this to just use `len(parts)` instead of a separate index, which I believe is equivalent.